enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -2270,6 +2270,11 @@ func (s *SchedulerServer) GetExecutionNodes(ctx context.Context, req *scpb.GetEx
 						(s.forceUserOwnedWindowsExecutors && isWindowsExecutor))),
 		}
 	}
+	slices.SortFunc(executors, func(a, b *scpb.GetExecutionNodesResponse_Executor) int {
+		aHID := a.GetNode().GetExecutorHostId()
+		bHID := b.GetNode().GetExecutorHostId()
+		return strings.Compare(strings.ToLower(aHID), strings.ToLower(bHID))
```

Comment:
wdyt about this:

```suggestion
		// Compare by host, then executor ID.
		if c := strings.Compare(a.GetHost(), b.GetHost()); c != 0 {
			return c
		}
		return strings.Compare(a.GetExecutorId(), b.GetExecutorId())
```

`host` is more human-friendly and I'd think people would want to sort by that, and `executor_id` is a guaranteed tie-breaker since it can't be overridden.